aws-cs-fargate/Infra/Program.cs

```diff
@@ -0,0 +1,148 @@
+                    new Ec2.Inputs.SecurityGroupIngressArgs {
+                        Protocol = "tcp",
+                        FromPort = 80,
+                        ToPort = 80,
+                        CidrBlocks = new[] { "0.0.0.0/0" },
+                    },
+                },
+            });
+
+            // Create an ECS cluster to run a container-based service.
+            var cluster = new Ecs.Cluster("app-cluster");
+
+            // Create an IAM role that can be used by our service's task.
+            var taskExecRole = new Iam.Role("task-exec-role", new Iam.RoleArgs {
+                AssumeRolePolicy = @"{
+    ""Version"": ""2008-10-17"",
+    ""Statement"": [{
+        ""Sid"": """",
+        ""Effect"": ""Allow"",
+        ""Principal"": {
+            ""Service"": ""ecs-tasks.amazonaws.com""
+        },
+        ""Action"": ""sts:AssumeRole""
+    }]
+}",
+            });
+            var taskExecAttach = new Iam.RolePolicyAttachment("task-exec-policy", new Iam.RolePolicyAttachmentArgs {
+                Role = taskExecRole.Name,
+                PolicyArn = "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy",
+            });
+
+            // Create a load balancer to listen for HTTP traffic on port 80.
+            var webLb = new Elb.LoadBalancer("web-lb", new Elb.LoadBalancerArgs {
+                Subnets = subnet.Ids,
+                SecurityGroups = new[] { webSg.Id },
+            });
+            var webTg = new Elb.TargetGroup("web-tg", new Elb.TargetGroupArgs {
+                Port = 80,
+                Protocol = "HTTP",
+                TargetType = "ip",
+                VpcId = vpc.Id,
+            });
+            var webListener = new Elb.Listener("web-listener", new Elb.ListenerArgs {
+                LoadBalancerArn = webLb.Arn,
+                Port = 80,
+                DefaultActions = new[] {
+                    new Elb.Inputs.ListenerDefaultActionsArgs {
+                        Type = "forward",
+                        TargetGroupArn = webTg.Arn,
+                    },
+                },
+            });
+
+            // Create a private ECR registry and build and publish our app's container image to it.
+            var appRepo = new Ecr.Repository("app-repo");
+            var appRepoCreds = appRepo.RegistryId.Apply(async rid => {
+                var creds = await Ecr.Invokes.GetCredentials(new Ecr.GetCredentialsArgs { RegistryId = rid });
+                var credsData = Convert.FromBase64String(creds.AuthorizationToken);
+                return Encoding.UTF8.GetString(credsData).Split(":");
+            });
+            var image = new Docker.Image("app-img", new Docker.ImageArgs {
+                Build = "../App",
+                ImageName = appRepo.RepositoryUrl,
+                Registry = new Docker.ImageRegistry {
+                    Server = appRepo.RepositoryUrl,
+                    Username = appRepoCreds.Apply(creds => creds[0]),
```

Comment:
I think you should be able to do `appRepoCreds.Get(0)`

Reply:
Hmm: `error CS1061: 'Output<string[]>' does not contain a definition for 'Get' and no accessible extension method 'Get' accepting a first argument of type 'Output<string[]>' could be found (are you missing a using directive or an assembly reference?)`

Reply:
My bad, should be `GetAt` https://github.com/pulumi/pulumi/blob/master/sdk/dotnet/Pulumi/Core/OutputExtensions.cs#L22